### PR TITLE
Follow upstream change to always check regular expression flags by `flags`

### PR DESCRIPTION
**Pull request by mathiasbynens**
https://github.com/tc39/ecma262/pull/2791

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -36887,13 +36887,11 @@ <h1>RegExp.prototype [ @@match ] ( _string_ )</h1>
           1. Let _rx_ be the *this* value.
           1. If Type(_rx_) is not Object, throw a *TypeError* exception.
           1. Let _S_ be ? ToString(_string_).
-          1. Let _global_ be ! ToBoolean(? Get(_rx_, *"global"*)).
-          1. If _global_ is *false*, then
+          1. Let _flags_ be ? ToString(? Get(_rx_, *"flags"*)).
+          1. If _flags_ does not contain *"g"*, then
             1. Return ? RegExpExec(_rx_, _S_).
           1. Else,
-            1. Assert: _global_ is *true*.
```
Comment (mathiasbynens) [inline spec.html:36894]: This assertion is no longer there upstream: https://tc39.es/ecma262/#sec-regexp.prototype-@@match

File: spec.html [hunk 2]
```diff
@@ -36959,11 +36957,11 @@ <h1>RegExp.prototype [ @@replace ] ( _string_, _replaceValue_ )</h1>
...
           1. If _functionalReplace_ is *false*, then
             1. Set _replaceValue_ to ? ToString(_replaceValue_).
-          1. Let _global_ be ! ToBoolean(? Get(_rx_, *"global"*)).
+          1. Let _flags_ be ? ToString(? Get(_rx_, *"flags"*)).
+          1. If _flags_ contains *"g"*, let _global_ be *true*. Otherwise, let _global_ be *false*.
           1. If _global_ is *true*, then
-            1. Let _fullUnicode_ be ! ToBoolean(? Get(_rx_, *"unicodeSets"*)).
-            1. If _fullUnicode_ is *false*, set _fullUnicode_ to ! ToBoolean(? Get(_rx_, *"unicode"*)).
             1. Perform ? Set(_rx_, *"lastIndex"*, *+0*<sub>𝔽</sub>, *true*).
+          1. If _flags_ contains *"u"* or _flags_ contains *"v"*, let _fullUnicode_ be *true*. Otherwise, let _fullUnicode_ be *false*.
```
Comment (gibson042) [inline spec.html:36964]: Decoupling assignment of _fullUnicode_ to no longer be conditioned on _global_ being true reads so much better!

**Review by gibson042 — APPROVED** [review]

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Comment by mathiasbynens** [comment]
I see the CI error, but somehow `npm run format` is a no-op for me locally. I'll go ahead and merge this, and we can deal with the formatting separately.

**Pull request merged**